.github/workflows/web_builds.yml

```diff
@@ -40,10 +40,11 @@ jobs:
       - uses: actions/checkout@v4
 
       - name: Set up Emscripten latest
-        uses: mymindstorm/setup-emsdk@v12
+        uses: mymindstorm/setup-emsdk@v14
         with:
           version: ${{env.EM_VERSION}}
           actions-cache-folder: ${{env.EM_CACHE_FOLDER}}
+          cache-key: ${{ matrix.cache-name }}-emsdk
```

Comment:
Oops, I forgot this variable only contains a prefix, and we actually add more to it when creating the actual cache name. So this works, but it's still not unique enough for multiple runs:

![image](https://github.com/godotengine/godot/assets/11782833/b421d043-b7e3-4806-9e84-7c9e662aa7e9)

This should match our main cache naming scheme.

```suggestion
          cache-key: ${{ matrix.cache-name }}-${{env.GODOT_BASE_BRANCH}}-${{github.ref}}-${{github.sha}}-emsdk
```

Reply:
Thanks, changed.

Overall, I wonder why caching is so convoluted. What we'd want typically is to just cache based on the Emscripten version, and that's it. But somehow it keeps trying to recreate the cache each time it's using it, instead of just using the version that's already there?

Reply:
Hm should the `-emsdk` part come earlier in the cache key? IIRC substrings are removed from right to left to find the closest cache hit. It should maybe be `emsdk-${{ matrix.cache-name }}-${{env.GODOT_BASE_BRANCH}}-${{github.ref}}-${{github.sha}}` instead?

Reply:
That's a good point. I guess if it's added at the end, we can accidentally download it when trying to find cache for scons itself.

It's not that substrings are removed, we do it manually with `restore-keys` (and `setup-emsdk` doesn't, AFAICT). But we do indeed risk a collision because the first part is the same. So yeah, should probably add it somewhere earlier, or at the very start. Good catch!